store.go

```diff
@@ -71,6 +71,20 @@ type metadataStore interface {
 	rwMetadataStore
 }
 
+// PutLayerFromStagingDirectoryOptions contains options to pass to PutLayerFromStagingDirectory
```

Comment:
(Documenting which of the options are mandatory and which are optional would be nice.)